Add multi-organization scope support with metrics aggregation and user deduplication + Champions page

YoavLax opened the pull request:
This PR introduces support for aggregating GitHub Copilot metrics across multiple organizations, enabling unified visibility and analysis of Copilot usage for teams working across organizational boundaries.

🆕 New Features
Multi-Organization Scope: Added new multi-organization scope type to fetch and aggregate metrics from multiple GitHub organizations simultaneously
Environment Configuration: Introduced NUXT_PUBLIC_GITHUB_ORGS environment variable for comma-separated organization list configuration
Parallel Data Fetching: Implemented concurrent API calls to multiple organizations with intelligent caching per organization

📊 Data Aggregation & Deduplication
Metrics Aggregation: Created aggregateMetricsByDate() function to merge metrics across organizations by date
Smart User Counting: Uses MAX aggregation for total_active_users instead of SUM to prevent double-counting users who are active in multiple organizations
Seats Deduplication: Implemented deduplicateSeats() to merge seat assignments across organizations, keeping the most recent activity per user
Teams Organization Tracking: Extended team data to track which organization each team belongs to, enabling proper team comparisons across orgs

Example configuration: 
<img width="354" height="58" alt="image" src="https://github.com/user-attachments/assets/884b4947-dd15-42d5-9d19-1856ae4c1a99" />

Technical Highlights
Maintains backward compatibility with existing single-organization scope
Preserves per-organization caching for optimal performance
Conservative user counting approach (MAX vs SUM) ensures accurate metrics without inflation
Fully type-safe TypeScript implementation with updated serialization methods

ylax1 committed:
Add multi-organization scope support with aggregation and UI updates

ylax1 committed:
Use MAX for total_active_users when aggregating multi-org metrics to avoid double-counting

ylax1 committed:
Show average daily users in UI and compute daily averages in stats endpoint

Update AgentModeViewer labels, tooltips and table headers from "Total Users with Activity" to "Average Daily Users with Activity" and clarify tooltip text. Calculate daily averages in server/api/github-stats.ts for overall user totals and per-model totals (divide summed counts by number of days) so UI and API consistently report average daily engaged users.

ylax1 committed:
Display average daily active users and add Avg Acceptance Rate card

ylax1 committed:
Add ChampionsViewer component and wire champions tab into MainComponent

- Add new ChampionsViewer.vue providing a Top 5 Teams card and an All Teams leaderboard
  with search, sorting, team detail links and visual rank badges.
- Load teams and per-team metrics, compute average acceptance rate and weekday avg active users,
  and support multi-organization display names.
- Integrate component into MainComponent: import ChampionsViewer, add 'champions' tab and
  render the viewer in the v-window.

ylax1 committed:
Show total suggestions/acceptances in Champions leaderboard and use cumulative totals for acceptance rate

- Add Total Suggestions and Total Acceptances columns to ChampionsViewer table (with icons and locale-formatted counts)
- Include totalSuggestions/totalAcceptances in TeamWithMetrics and populate when loading team metrics
- Change acceptance rate calculation in ChampionsViewer to use cumulative totals (total_acceptances / total_suggestions) and return totals from loadMetricsForTeam
- Update TeamsComponent average acceptance rate computation to use cumulative totals per team (consistent with MetricsViewer)

karpikpl commented:
would it be possible just to re-use `NUXT_PUBLIC_GITHUB_ORG` and do multi-orgs when field has comma separated list?

getting metrics on enterprise level doesn't already provide org aggregated metrics?

ylax1 committed:
Consolidate multi-org env var to NUXT_PUBLIC_GITHUB_ORG (comma-separated)

- Update README to document NUXT_PUBLIC_GITHUB_ORG as single or comma-separated list and remove references to NUXT_PUBLIC_GITHUB_ORGS
- Parse comma-separated githubOrg in Options runtime config to populate githubOrgs and enable multi-organization scope
- Remove githubOrgs runtime config entry from nuxt.config.ts (use githubOrg for both single and multi-org)
- Update getDisplayName to derive org list from githubOrg for multi-organization scope

lvthillo commented:
Any update on this PR?
